pkgs/tools/security/nwaku/default.nix

```diff
@@ -0,0 +1,97 @@
+{
+  stdenv, lib, fetchurl, fetchFromGitHub,
+  nim, cmake, which, writeScriptBin,
+  # Options: wakunode1, wakunode2, wakubridge
+  makeTargets ? [ "wakunode2" ],
+  # WARNING: CPU optmizations that make binary not portable.
+  nativeBuild ? false
+}:
+
+assert nim.version == "1.6.12";
+
```

Comment:
```suggestion
```

Reply:
No, that is staying there. The Nim compiler version check is necessary to ensure stability.

Reply:
This will break when nim is updated and will likely be removed by the next person because of that

Reply:
Then I can lock Nim version at `all-packages.nix` level as I do for `nimbus-beacon-node` package.

Reply:
Is this really necessary? Are nim updates really that unstable? Nixpkgs is a distribution and normally a patch update for a compiler should not break all applications and we would normally apply that to all programs.

Reply:
Yes. Status as a company has probably the most Nim developers, including Nim compiler contributors, among its staff than any other organization, and yes, the use of specific Nim compiler versions is required to avoid very weird issues.

I'll get to locking it up later this week.